src/lib/middleware/bearer-token-middleware.ts

```diff
@@ -0,0 +1,24 @@
+import type { Request, Response, NextFunction } from 'express';
+import type { IUnleashConfig } from '../types';
+
+export const bearerTokenMiddleware = ({
+    getLogger,
+    flagResolver,
+}: Pick<IUnleashConfig, 'getLogger' | 'flagResolver'>) => {
+    const logger = getLogger('/middleware/bearer-token-middleware.ts');
+    logger.debug('Enabling bearer token middleware');
+
+    return (req: Request, _: Response, next: NextFunction) => {
+        if (flagResolver.isEnabled('bearerTokenMiddleware')) {
```

Comment:
We should have it enabled for signals as it's an experimental endpoint and we do want to support Bearer tokens in this endpoint, while the others we want to have it as a release flag. Should there be any security related issue with Bearer tokens in signals we can just kill the whole functionality, and that should be fine because it's experimental

Reply:
Addressed in https://github.com/Unleash/unleash/pull/6624/commits/9a558dffd6b76baa51191ba18510c0cc2e0cc2cf